internal/scm-clients/gitlab/gitlab.go

```diff
@@ -30,12 +30,11 @@ type GitlabClientImpl struct {
 
 var _ GitlabClient = (*GitlabClientImpl)(nil) // Verify that *GitlabClientImpl implements gitlabClient.
 
-func InitClient(client *http.Client, token string) (GitlabClient, error) {
-	gc, _ := gitlab.NewClient(token, gitlab.WithHTTPClient(client))
+func InitClient(client *http.Client, token string, host string) (GitlabClient, error) {
+	gc, _ := gitlab.NewClient(token, gitlab.WithHTTPClient(client), gitlab.WithBaseURL(fmt.Sprintf("https://%s/api/v4", host)))
```

Comment:
I think we can support this new capability as an optional parameter, without breaking the current behavior

Reply:
I've attempted to add support without breaking current behaviour, although haven't found an elegant way to make it optional (I'm not a Go development so might have overlooked something). Open for suggestions.

Reply:
what about doing the same approach as when init github, with checking the host value and calling the right `NewClient`?